demos/python_demos/machine_translation_demo/machine_translation_demo.py

```diff
@@ -0,0 +1,229 @@
+            tokens (np.array): sentence in tokenized format.
+            remove_repeats (bool): remove repeated words.
+
+        Returns:
+            sentence (str): output sentence.
+        """
+        sentence = self.tokenizer.decode(tokens)
+        for s in self.idx.keys():
+            sentence = sentence.replace(s, '')
+        if remove_repeats:
+            sentence = self._remove_repeats(sentence)
+        return sentence.lstrip()
+
+    def _extend_tokens(self, tokens):
+        """ Extend tokens.
+
+        Arguments:
+            tokens (np.array): sentence in tokenized format.
+
+        Returns:
+            tokens (np.array): extended tokens.
+        """
+        tokens = [self.idx['<s>']] + tokens + [self.idx['</s>']]
+        pad_length = self.max_tokens - len(tokens)
+        if pad_length > 0:
+            tokens = tokens + [self.idx['<pad>'] for i in range(pad_length)]
+        return tokens
+
+    def _remove_repeats(self, sentence):
+        """ Remove repeated words.
+
+        Arguments:
+            sentence (str): sentence.
+
+        Returns:
+            sentence (str): sentence in lowercase without repeated words.
+        """
+        tokens = sentence.lower().split()
+        tokens_new = []
+        if len(tokens):
+            tokens_new = [tokens[0]]
+            for i in range(1, len(tokens)):
+                if tokens[i] != tokens_new[-1]:
+                    tokens_new.append(tokens[i])
+        return " ".join(tokens_new)
+
+
+def build_argparser():
+    """ Build argument parser.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-m", "--model", required=True, type=str,
+                        help="Required. Path to an .xml file with a trained model")
+    parser.add_argument('--tokenizer-src', type=str, required=True,
+                        help='Required. Path to the folder with src tokenizer that contains vocab.json and merges.txt.')
+    parser.add_argument('--tokenizer-tgt', type=str, required=True,
+                        help='Required. Path to the folder with tgt tokenizer that contains vocab.json and merges.txt.')
+    parser.add_argument('--output-name', type=str, default='pred/Squeeze',
+                        help='Name of the models output node.')
+    return parser
+
+
+def main(args):
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+    log.info("creating translator")
+    model = Translator(
+        model_xml=args.model,
+        model_bin=os.path.splitext(args.model)[0] + ".bin",
+        tokenizer_src=args.tokenizer_src,
+        tokenizer_tgt=args.tokenizer_tgt,
+        output_name=args.output_name
+    )
+    log.info("enter 'q!' to exit.")
+    while True:
+        sentence = input("> ")
+        if sentence == "q!":
+            break
+        try:
+            start = time.time()
+            translation = model(sentence)
+            stop = time.time()
+            log.info(translation)
```

Comment:
I don't think you should _log_ the translation - it's not just some message, it's the output of the program. You should just print it.

Reply:
done